File: Element/Remove_Duplicates.py
# Remove Duplicates from Sorted Array
# O(N) time complexity

def rem(arr)->int:
	lo = 0
	hi = lo+1
	while(lo < hi and hi < len(arr)):
		if(arr[lo]==arr[hi]):
			arr.remove(arr[lo])
			lo+=1
			hi+=1
		else:
			lo+=1
			hi+=1
	return len(arr)

if __name__=='__main__':
	arr = [1,1,2,2,3]
	print(rem(arr),end="\n") # 3
	

File: Element/Remove_Duplicates.cxx
// Remove Duplicates from a Sorted Array
// O(N) time complexity

#include <bits/stdc++.h>
using namespace std;

class Solution{
	public:
	static int rem(vector<int> arr){
		arr.erase(unique(arr.begin(),arr.end()),arr.end());
		return arr.size();
	}
};

int main(){
	ios_base::sync_with_stdio(0);
	cin.tie(0); cout.tie(0);
	vector<int> arr = {1,1,1,2,2,3};
	cout << Solution::rem(arr) << endl; // 3
	return 0;
}
		


